Framework/PythonInterface/mantid/simpleapi.py

```diff
@@ -302,6 +302,39 @@ def fitting_algorithm(f):
     When applied to a function definition this decorator replaces its code with code of
     function 'wrapper' defined below.
     """
+    def gather_fitting_returns(function_name, lhs, algm):
```

Comment:
In my opinion, this is not an error of the Python layer, but rather in the design of Fit. There really should not be any 'temporary' output properties in any algorithm. The optional output workspaces should have been declared in `Fit::init()` as non-mandatory properties, not post-mortem in `Fit::createOutput()`.

Anyway, as you say, this change is limited to the Fit specific part of simpleapi, and I'm not feeling too strongly about it. Most importantly, the Python API breakage gets fixed, so I'm OK with this.